### DISCUSSION TITLE
想问一下agent的内存占用情况的疑惑

### DISCUSSION THREAD

**Discussion by zcl1115** [harmonycloud/kindling#627] [agent collector] [answered]
agent启动后占用内存会逐步上涨，最终会稳定要一个动态平衡的状态，想问一下这个内存占用的大小会跟什么有关，从内核采集的事件数量吗，因为这边看到某些机器agent上处理的事件数并不多，但是内存最终占用非常大，如果我想要对agent占据的内存做内存分析，使用什么好一点，这边用pprof观察好像看到inuse_space是很小的，跟collector本身占用的内存对不上

**Comment by dxsup** [comment, marked as answer]
因为agent使用了cgo调用C代码，导致有部分内存无法被pprof统计到，cgo部分包括kindling获取内核事件和通过netlink获取conntrack表信息。主要的内存占用是cgo部分和collector的指标数据。

要分析内存可以尝试jemalloc这个内存分配器，可以统计到C部分申请的内存。
